ibis/pandas/core.py

```diff
@@ -55,7 +55,7 @@ def find_data(expr):
             seen.add(node)
 
             if hasattr(node, 'source'):
-                data[node] = node.source.dictionary[node.name]
+                data[node] = node.source.dictionary.get(node.name, None)
```

Comment:
Don't need to do this now, but might be worth thinking about whether we can detect if a backend hasn't provided data and provides neither a `pre_execute` nor `data_preload` implementations and raise an error.